File: Data Structure Using Java/Searching/BinarySearch.java
public class BinarySearch {

    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (arr[mid] == target)
                return mid;
            else if (arr[mid] < target)
                left = mid + 1;// Discard left half
            else
                right = mid - 1; // Discard right half
        }
        return -1; // if the target is not found then we are returning -1
    }

    public static void main(String[] args) {
        int[] arr = { 1, 3, 7, 9, 11, 12, 45, 48, 61, 65 };
        int target = 65;
        System.out.printf("The %d is present at %d", target, binarySearch(arr, target));
    }
}
// Note: For Binary Search to work best the array should be sorted
// Time complexity T(n) = O(log n)

File: Data Structure Using C/stack/binary_search.c
#include <stdio.h>

int binary_search(int arr[], int low, int high, int target)
{
    while (low <= high)
    {
        int mid = low + (high - low) / 2;
        if (target > arr[mid])
        {
            low = mid + 1;
        }
        else if (target < arr[mid])
        {
            high = mid - 1;
        }
        else
        {
            return mid;
        }
    }

    return -1;
}

int main()
{
    int arr[10] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int target;
    int l = 0;
    int r = 9;
    scanf("%d", &target);
    int index = binary_search(arr, l, r, target);

    if (index >= 0)
    {
        printf("Found at index %d\n", index);
    }
    else
    {
        printf("Not found\n");
    }

    return 0;
}